report/src/main/java/com/megaease/easeagent/report/sender/metric/MetricKafkaSender.java

```diff
@@ -102,14 +102,12 @@ public void close() throws IOException {
     }
 
     private void initAppenderManager() {
-        if (appenderManager != null) {
-            return;
-        }
-        synchronized (MetricKafkaSender.class) {
-            if (appenderManager != null) {
-                return;
+        if (appenderManager == null) {
+            synchronized (MetricKafkaSender.class) {
+                if (appenderManager == null) {
+                    appenderManager = AppenderManager.create(this.outputProperties);
+                }
```

Comment:
```java
if(xxx!=null){
    return xxx;
}
//init xxx
if(xxx!=null){
   return xxx;
}

//init xxx
if(xxx!=null){
   return xxx;
}
```
```java
if(xxx==null){
   //init xxx
   if(xxx==null){
      //init xxx  
       if(xxx==null){
          //init xxx
       }
   }
}
return xxx;
```
The first is more readable than the second, because the first is less nested.
At the same time they achieve the same purpose, so I don't understand why you modify this piece of code

Reply:
When dealing with multiple nesting levels, ` return xxx` It's really clearer.
Modified based on the following two points:

- The nesting level of `Double-check` is not very deep.
- Personal habits

If necessary, I'll revert and rebase it.

Reply:
> When dealing with multiple nesting levels, ` return xxx` It's really clearer. Modified based on the following two points:
> 
> * The nesting level of `Double-check` is not very deep.
> * Personal habits
> 
> If necessary, I'll revert and rebase it.

I mean it's not necessary. If the original code is better, it doesn't make much sense to change it in your PR.
Of course, you add volatile as a good defense, it's a good operation